modules/minimp3/audio_stream_mp3.cpp

```diff
@@ -239,7 +239,7 @@ void AudioStreamMP3::set_data(const Vector<uint8_t> &p_data) {
 }
 
 Vector<uint8_t> AudioStreamMP3::get_data() const {
-	return data;
+	return Vector<uint8_t>(data);
```

Comment:
There seems to be a conversion from `Vector` -> `LocalVector` right above in `set_data`. Is that direction not handled in this PR?

Because here it seems there's just no point for `data` to be `LocalVector` if the input data is `Vector` and the output should be too. (Unless I guess we're doing stuff with `data` where `LocalVector` really gives a huge benefit and warrants this double conversion.)

Reply:
We do this in the codebase sometimes, where we store `LocalVector` internally but expose `Vector` interfaces.
The reason is that `Vector` still has considerable overhead, even on lookup. If we expect user queries to happen rarely, but internal read/writes to occur often, this type of interface is still preferable, even though we're re-allocating for user queries.